plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -92,16 +94,24 @@ class FeatureDevSessionContext(val project: Project) {
 
     suspend fun zipFiles(projectRoot: VirtualFile): File = withContext(getCoroutineBgContext()) {
         val files = mutableListOf<VirtualFile>()
+        var totalSize: Long = 0
+
         VfsUtil.visitChildrenRecursively(
             projectRoot,
             object : VirtualFileVisitor<Unit>() {
                 override fun visitFile(file: VirtualFile): Boolean {
-                    if (file.isFile) {
-                        files.add(file)
-                        return true
-                    }
                     return runBlocking {
-                        !ignoreFile(file, this)
+                        if (file.isFile && !ignoreFile(file, this)) {
+                            totalSize += file.length
+                            files.add(file)
+
+                            if (maxProjectSizeBytes != null && totalSize > maxProjectSizeBytes) {
+                                throw RepoSizeLimitError(message("amazonqFeatureDev.content_length.error_text"))
```

Comment:
question: did you check if this is not breaking another team's functionality?

Reply:
if `maxProjectSizeBytes` isn't send to this class it will not do anything